x-pack/filebeat/input/awss3/_meta/terraform/README.md

```diff
@@ -0,0 +1,43 @@
+# Terraform setup for AWS S3 Input Integration Tests
+
+This directory contains a Terrafrom module that creates the AWS resources needed
+for executing the integration tests for the `aws-s3` Filebeat input. It creates
+an S3 bucket and SQS queue and configures S3 `ObjectCreated:*` notifications to
+be delivered to SQS.
+
+It outputs configuration information that is consumed by the tests to
+`outputs.yml`. The AWS resources are randomly named to prevent name collisions
+between multiple users.
+
+### Usage
+
+You must have the appropriate AWS environment variables for authentication set
+before running Terraform or the integration tests. The AWS key must be
+authorized to create and destroy S3 buckets and SQS queues.
+
+1. Execute terraform in this directory to create the resources. This will also
+write the `outputs.yml`.
+
+    `terraform apply`
+
+2. (Optional) View the output configuration.
+
+   ```yaml
+   "aws_region": "us-east-1"
+   "bucket_name": "filebeat-s3-integtest-8iok1h"
+   "queue_url": "https://sqs.us-east-1.amazonaws.com/144492464627/filebeat-s3-integtest-8iok1h"
+   ```
+
+2. Execute the integration test.
+
+    ```
+    cd x-pack/filebeat/inputs/awss3
+    go test -tags aws,integration -run TestInputRun -v .
```

Comment:
@kaiyan-sheng would this work with our existing AWS ci integration? that would be great.

Reply:
I think we have the aws tests run weekly right now in the CI and this would work. @v1v Do you know if this is easy to add?

Reply:
The AWS tests run on a weekly basis, as you said, and it runs the commands defined in:
- https://github.com/elastic/beats/blob/d8985337bc81e8f33efe6458ef202983c42d99d6/x-pack/metricbeat/Jenkinsfile.yml#L35-L47

`mage build test` runs in the folder `x-pack/metricbeat` so, if the above-mentioned command is part of the mage then it should work.

Reply:
the test requires applying a terraform file (and destroying it at the end):
https://github.com/elastic/beats/blob/master/x-pack/filebeat/input/awss3/_meta/terraform/README.md#usage

will this happen?

Reply:
btw, shouldn't be the file https://github.com/elastic/beats/blob/master/x-pack/filebeat/Jenkinsfile.yml?